### ISSUE TITLE
Make DefaultCookieName a static readonly field instead of a property

### ISSUE THREAD

**Issue by hishamco** [aspnet/Localization#36]
If we have look to this line of code https://github.com/aspnet/Localization/blob/dev/src/Microsoft.AspNet.Localization/CookieRequestCultureProvider.cs#L47, It will be nice if declare ```DefaultCookieName``` as constant rather than read only property, because the value will never change.
```csharp

public static readonly string DefaultCookieName = "ASPNET_CULTURE";
```

**Comment by hishamco** [comment]
\cc @dougbu

**Comment by dougbu** [comment]
I'll leave this to triage to make a call.

**Comment by hishamco** [comment]
\cc @DamianEdwards

**Comment by hishamco** [comment]
@DamianEdwards can you have a look to this if you don't mind

**Comment by hishamco** [comment]
Please let me know if a PR is needed for this change (if it's accepted)

**Comment by kirthik** [comment]
cc @Eilon, @danroth27 for triaging.

**Comment by Eilon** [comment]
This should remain a `static readonly` because it is not a universal truth, and it certainly *could* change.

**Comment by hishamco** [comment]
Oops!! 
@Eilon that's why I did, have a look https://github.com/aspnet/Localization/pull/51/files#diff-fbf4e39686d45658872d8ef10043d143L47

**Comment by Eilon** [comment]
Oh sorry, the bug and PR say `const` but the change is to make it `static readonly`.

**Comment by hishamco** [comment]
No problem, but the tiltle is **Make DefaultCookieName a static readonly field instead of a property** perhaps the commit title :smile:

**Comment by Eilon** [comment]
I just edited the title :smile:

**Comment by hishamco** [comment]
lol BTW I meant be ```static readonly``` as constant :smile:

**Comment by Eilon** [comment]
Ok, but just to be clear, `static readonly` is completely different from `const`.

**Comment by hishamco** [comment]
Yep, but while the value will never change ```const``` will be equivalent with ```static readonly``` while the intend will never change the ```DefaultCookieName``` in the constructor

**Comment by Eilon** [comment]
There is no guarantee that the cookie name won't change at some point in the future. We use `const` only for universal truths because the laws of the universe don't change (I hope!).

A `const` value gets baked into every usage, which means that if you later change the value of the `const` any already-compiled code will continue to use the old value. A `static readonly` is always established at first usage, and so the value can be changed in a newer build/version of the DLL.

Also, `const` values can be used in places where `const` values are required, such as in attribute values.

We might change the default cookie name tomorrow and we want to make sure existing compiled libraries that use the value will not break.

**Comment by hishamco** [comment]
As you mentioned before (I hope!), regarding the ```static readonly``` i know it will established at first usage, but why already-compiled code will use the old value in case of ```const``` while there's a new build?!!

**Comment by Eilon** [comment]
@hishamco that's how .NET compilers work. A `const` is a `const`, so its value can (and often *must*) be embedded at the callsite.

**Comment by hishamco** [comment]
Thanks for clarification

**Comment by DamianEdwards** [comment]
I purposely made it *not* readonly so that the app could change it if it really wanted to.

**Comment by Eilon** [comment]
@DamianEdwards it is currently a read-only property, so it can't be changed to begin with...

**Comment by DamianEdwards** [comment]
Doh, right you are. Back in my box. We should make consistent with whatever we do in other places then.

**Comment by Eilon** [comment]
@DamianEdwards consistent with what?

**Comment by hishamco** [comment]
@DamianEdwards it was originally static string that assigned to read only property, so it will be handy to make it ```static readonly``` field

**Comment by kirthik** [comment]
merged c228b65d478561869daa9161fc97a89ff02cad2e
